apps/service_providers/speech_service.py

```diff
@@ -147,3 +148,33 @@ def transcribe_audio(self, audio: BytesIO) -> str:
                 if cancellation_details.error_details:
                     msg += f". Error details: {cancellation_details.error_details}"
             raise AudioSynthesizeException(msg)
+
+
+class OpenAISpeechService(SpeechService):
+    _type: ClassVar[str] = SyntheticVoice.OpenAI
+    supports_transcription: ClassVar[bool] = True
+    openai_api_key: str
+    openai_api_base: str = None
+    openai_organization: str = None
+
+    @property
+    def _client(self) -> OpenAI:
+        return OpenAI(api_key=self.openai_api_key, organization=self.openai_organization, base_url=self.openai_api_base)
+
+    def _synthesize_voice(self, text: str, synthetic_voice: SyntheticVoice) -> SynthesizedAudio:
+        """
+        Calls OpenAI to convert the text to speech using the synthetic_voice
+        """
+        response = self._client.audio.speech.create(model="tts-1", voice=synthetic_voice.name, input=text)
+        audio_data = response.read()
+
+        audio_segment = AudioSegment.from_file(BytesIO(audio_data), format="mp3")
+        duration_seconds = len(audio_segment) / 1000  # Convert milliseconds to seconds
+        return SynthesizedAudio(audio=BytesIO(audio_data), duration=duration_seconds, format="mp3")
+
+    def transcribe_audio(self, audio: BytesIO) -> str:
+        transcript = self._client.audio.transcriptions.create(
```

Comment:
we should probably have some error handling here

Reply:
I added https://github.com/dimagi/open-chat-studio/pull/340/commits/399845285b39eaa084af458b5adc4899756b9816. I think if we catch + raise an `AudioTranscriptionException` where we call it from the channel.py file, we should be fine. I'm wondering, we _can_ provide the user with feedback, like we do when the user sends an unsupported message. In this case, we'll just tell the user "listen, what you sent us isn't working. Please send something better". Obviously the bot will formulate an appropriate response here...

Reply:
One nice thing about doing the generic error handling in the service is that users of the service only need to catch the service exceptions and not base Exception.

Reply:
Yes, I mean doing something like this: https://github.com/dimagi/open-chat-studio/blob/main/apps/service_providers/speech_service.py#L39-L42

Where we catch and re-raise as an app specific exception. The benefit is that callers only have to care about that one exception and not any others.

Reply:
Yeah that makes sense

Reply:
https://github.com/dimagi/open-chat-studio/pull/340/commits/ef9c0ff55216740f577bcfc159b3c1801f981139